legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt

```diff
@@ -0,0 +1,14 @@
+package com.hamdan.forzenbook.legacy.core.view
+
+class NavigatorImpl(
+    val routeToLogin: () -> Unit,
```

Comment:
This IS the impl. An impl that takes in lambdas with the true impl doesn't make sense lol it's fine for this class to take in a context in its methods and route by itself

Reply:
ok that makes sense, I hope that's the only last step in that implementation concept that I'm missing considering how the struggle was trying to understand implementing the concept.